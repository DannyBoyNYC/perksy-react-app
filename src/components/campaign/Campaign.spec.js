const React = require('react');

const Campaign = require('./index').default;

describe('<Campaign.* />', () => {
  let component;
  const requiredProps = {
    name: 'Campaign Name',
    respondents: {
      current: 898,
      maximum: 1112
    }
  };

  describe('<Campaign.Card />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Campaign.Card {...requiredProps} />,
        'Campaign'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should render children if passed', () => {
      component.setProps({
        children: <div>child content</div>
      });
      expect(component).toIncludeText('child content');
    });

    it('should render an Image.Lazy', () => {
      expect(component).toContainExactlyOneMatchingElement('Lazy');
    });
  });

  describe('<Campaign.Percentage />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Campaign.Percentage {...requiredProps} />,
        'Campaign'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should render a ProgressBar.Percentage', () => {
      expect(component).toContainExactlyOneMatchingElement('Percentage');
    });
  });

  describe('<Campaign.Progress />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Campaign.Progress {...requiredProps} />,
        'Campaign'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should render a ProgressBar.Basic', () => {
      expect(component).toContainExactlyOneMatchingElement('Basic');
    });
  });

  describe('<Campaign.Row />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Campaign.Row {...requiredProps} />,
        'Campaign'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should render an Image.Lazy', () => {
      expect(component).toContainExactlyOneMatchingElement('Lazy');
    });
  });
});
