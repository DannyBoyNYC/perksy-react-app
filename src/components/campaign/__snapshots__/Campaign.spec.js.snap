// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Campaign.* /> <Campaign.Card /> should render correctly with all required props 1`] = `
<div
  className="Campaign-root-0-1-1 Campaign-card-0-1-6"
>
  <Lazy
    className="Campaign-image-0-1-3"
  />
  <div
    className="Campaign-name-0-1-4"
  >
    Campaign Name
  </div>
</div>
`;

exports[`<Campaign.* /> <Campaign.Percentage /> should render correctly with all required props 1`] = `
<div
  className="Campaign-root-0-1-1 Campaign-percentage-0-1-7"
>
  <div
    className="Campaign-name-0-1-4"
  >
    Campaign Name
  </div>
  <div
    className="Campaign-progress-0-1-8"
  >
    <Percentage
      current={898}
      maximum={1112}
    />
  </div>
</div>
`;

exports[`<Campaign.* /> <Campaign.Progress /> should render correctly with all required props 1`] = `
<div
  className="Campaign-root-0-1-1 Campaign-progress-0-1-8"
>
  <div
    className="Campaign-name-0-1-4"
  >
    Campaign Name
  </div>
  <div
    className="Campaign-progress-0-1-8"
  >
    <Basic
      current={898}
      maximum={1112}
    />
  </div>
</div>
`;

exports[`<Campaign.* /> <Campaign.Row /> should render correctly with all required props 1`] = `
<div
  className="Campaign-root-0-1-1 Campaign-row-0-1-9"
>
  <Lazy
    className="Campaign-image-0-1-3"
  />
  <div
    className="Campaign-name-0-1-4"
  >
    Campaign Name
  </div>
  <div
    className="Campaign-spacer-0-1-5"
  />
</div>
`;
